src/libstd/thread/mod.rs

```diff
@@ -533,12 +538,20 @@ impl Thread {
     pub fn name(&self) -> Option<&str> {
         self.inner.name.as_ref().map(|s| &**s)
     }
+
+    /// Gets the thread's unique ID.
+    ///
+    /// This ID is unique across running threads, and a thread that has stopped
+    /// might see its ID reused.
+    pub fn id(&self) -> u32 {
```

Comment:
If this PR is accepted, this will need to be marked `#[unstable]` and have a tracking issue opened. (See `catch_panic` in this module for an example.)